metricflow-semantics/metricflow_semantics/specs/non_additive_dimension_spec.py

```diff
@@ -33,6 +34,13 @@ class NonAdditiveDimensionSpec(SerializableDataclass):
     window_choice: AggregationType
     window_groupings: Tuple[str, ...] = ()
 
+    def __post_init__(self) -> None:
+        """Post init validator to ensure names with double-underscores are not allowed."""
+        assert self.name.find(DUNDER) == -1, (
```

Comment:
Did you already check that this is actually handled in validation? If not I would be worried this might be a braking change.

Reply:
I did, but it's kind of roundabout. This value has to be a name matching the name attribute of a Dimension object in the same semantic model as the measure. So it can't even be metric_time, no matter what our docs examples show.

Reply:
So you're not worried it could be a breaking change? If you are at all worried, I'd recommend making this a logged error instead of raised like the other ones mentioned in this PR! But if not then ignore me